src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/Context/FormattingContext.cs

```diff
@@ -235,15 +235,29 @@ public void AddIndentBlockOperation(IndentBlockOperation operation)
             }
 
             // get indentation based on its previous indentation
-            var indentationGetter = new Lazy<int>(() =>
+            if (indentationData is not AdjustedIndentationData { BaseIndentationData: var baseIndentationData, Adjustment: var totalAdjustment })
             {
-                var indentationDelta = operation.IndentationDeltaOrPosition * Options.IndentationSize;
+                (baseIndentationData, totalAdjustment) = (indentationData, 0);
+            }
 
-                return indentationData.Indentation + indentationDelta;
```

Comment:
📝 `indentationData.Indentation` could be a computed property against values that change during formatting. The result of this computation is no longer directly cached.